### ISSUE TITLE
feat: Learning rate scheduler support (ReduceOnPlateau, CosineAnnealing)

### ISSUE THREAD

**Issue by dndungu** [zerfoo/zerfoo#101]
## Summary

Training tabular models on noisy financial data benefits greatly from learning rate scheduling. A fixed LR either converges too slowly (low LR) or oscillates past minima (high LR). Scheduling adapts the LR during training.

## Priority Schedulers

1. **ReduceOnPlateau** — reduce LR when validation loss stops improving. Most practical for financial data where convergence is unpredictable.
2. **CosineAnnealing with warm restarts** — already tracked in #48, seconding the request.

## Proposed API

```go
// In package training/scheduler
type Scheduler[T numeric.Float] interface {
    Step(epoch int, metric float64) // update LR based on epoch or metric
    GetLR() T
}

type ReduceOnPlateau[T numeric.Float] struct {
    Factor   T       // LR multiplied by this on plateau (default 0.1)
    Patience int     // epochs to wait before reducing (default 10)
    MinLR    T       // minimum LR (default 1e-6)
}

// Usage with optimizer
scheduler := scheduler.NewReduceOnPlateau[float32](0.1, 10, 1e-6)
for epoch := range epochs {
    loss := trainEpoch(...)
    scheduler.Step(epoch, float64(loss))
    adamw.SetLR(scheduler.GetLR())
}
```

## Use Case

Wolf trains MLP on 200K+ minute-bar rows. With fixed LR=0.001, training loss plateaus after ~30 epochs while validation loss diverges. ReduceOnPlateau would automatically lower LR at the plateau, allowing the model to converge to a tighter minimum.

## Related

- #48 (CosineAnnealing — complementary)
- #47 (early stopping — often used together with LR scheduling)
